tedana/utils.py

```diff
@@ -41,6 +41,45 @@ def load_image(data):
     return fdata
 
 
+def extreme_percentile_mask(dd, ees=None):
+    """
+    Returns mask for data between [0.001, 5] * 98th percentile of dd
+
+    Parameters
+    ----------
+    dd : (S x E x T) array_like
+        Input data, where `S` is samples, `E` is echos, and `T` is time
+    ees : (N,) :obj:`list`
+        Indices of echos to assess from `dd` in calculating output
+
+    Returns
+    -------
+    imask : (S x N) :obj:`numpy.ndarray`
+        0 1 int array denoting
+    """
+
+    if ees is None:
+        ees = range(dd.shape[1])
+    imask = np.zeros((dd.shape[0], len(ees)), dtype=bool)
+    for ee in ees:
+        if len(ees) == 1:
+            LGR.debug('Creating eimask for optimal combination')
+        else:
+            LGR.debug('Creating eimask for echo {}'.format(ee))
+        perc98 = stats.scoreatpercentile(dd[:, ee, :].flatten(), 98,
+                                         interpolation_method='lower')
+        lthr, hthr = 0.001 * perc98, 5 * perc98
+        LGR.debug('Eimask threshold boundaries: '
+                  '{:.03f} {:.03f}'.format(lthr, hthr))
+        m = dd[:, ee, :].mean(axis=1)
+        imask[np.logical_and(m > lthr, m < hthr), ee] = True
+
+    # if any echo has an outlier set the mask for that voxel to False
+
+    mask = imask.all(axis=1).astype(int)
```

Comment:
The code says "all" but the comment says "any". I noticed in the conversation that you were leaning toward any. Is that still accurate?

Reply:
Permissions should now be changed. FWIW, this is still a work in progress. My hope is to get #358 merged and then get this working on that code.